ui/src/features/project/settings/views/general/general-settings.tsx

```diff
@@ -1,41 +1,82 @@
+import { Code, ConnectError } from '@connectrpc/connect';
 import { useMutation, useQuery } from '@connectrpc/connect-query';
 import { zodResolver } from '@hookform/resolvers/zod';
-import { Button, Card, Flex, message } from 'antd';
+import { Button, Card, Flex, message, notification } from 'antd';
 import type { JSONSchema4 } from 'json-schema';
 import { useForm } from 'react-hook-form';
 import { useParams } from 'react-router-dom';
 import yaml from 'yaml';
 import { z } from 'zod';
 
+import { newErrorHandler, newTransportWithAuth } from '@ui/config/transport';
 import { YamlEditor } from '@ui/features/common/code-editor/yaml-editor';
 import { FieldContainer } from '@ui/features/common/form/field-container';
 import {
+  projectConfigYAMLExample,
+  projectYAMLExample
+} from '@ui/features/project/list/utils/project-yaml-example';
+import {
+  createOrUpdateResource,
   getProject,
+  getProjectConfig,
   updateResource
 } from '@ui/gen/api/service/v1alpha1/service-KargoService_connectquery';
 import { RawFormat } from '@ui/gen/api/service/v1alpha1/service_pb';
+import projectConfigSchema from '@ui/gen/schema/projectconfigs.kargo.akuity.io_v1alpha1.json';
 import schema from '@ui/gen/schema/projects.kargo.akuity.io_v1alpha1.json';
 import { decodeRawData } from '@ui/utils/decode-raw-data';
 import { zodValidators } from '@ui/utils/validators';
 
 const formSchema = z.object({
   value: zodValidators.requiredString
 });
-import { projectYAMLExample } from '../../../list/utils/project-yaml-example';
 
 import { DeleteProject } from './delete-project-modal';
 
+const transport = newTransportWithAuth(
+  newErrorHandler((err) => {
+    if (err.code === Code.NotFound) {
+      // ignore
+      return;
+    }
+
+    const errorMessage = err instanceof ConnectError ? err.rawMessage : 'Unexpected API error';
+    notification.error({ message: errorMessage, placement: 'bottomRight' });
+  })
+);
```

Comment:
Can you move this into a separate file?
Also, please add a comment on why we need this.